### PR TITLE
HDDS-10214. Update supported versions in security policy up to 1.4.0

### PR DESCRIPTION
**Pull request by ivandika3**
## What changes were proposed in this pull request?

Currently Security.md only contains Ozone version up to 1.1

This patches include version up to 1.4.

It is also a good idea to add information to the Ozone wiki release guide to update the security.md during the release process.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10214

## How was this patch tested?

NA

### CODE REVIEW

**Comment by adoroszlai** [comment]
Thanks @ivandika3 for updating this doc.  Adding new versions is definitely needed.

1.2.0 should be omitted, since it is superseded by 1.2.1.

More importantly, I think we should discuss (probably in the dev mailing list) which GA versions are really supported.  With the exception of 1.2.1, there were no patch releases with security fixes.

**Comment by ivandika3** [comment]
@adoroszlai Thank you for review this.

> 1.2.0 should be omitted, since it is superseded by 1.2.1.

Updated.

> I think we should discuss (probably in the dev mailing list) which GA versions are really supported. With the exception of 1.2.1, there were no patch releases with security fixes.

I see, I'm not quite clear what does "supported" really means. My understanding is that people can submit security reports only for the supported versions.

**Review by fapifta — COMMENTED** [review]
Hi @ivandika3 thank you for dealing with the documentation in this PR.

I did a quick search for CVEs, as we had a few in the past, according to this page:
https://www.opencve.io/cve?vendor=apache&product=ozone

we have fixed our CVEs in 1.2.0 so I suggest to support releases after 1.2.1, as 1.2.1 also is released to fix a security problem.

I think we also miss 1.2.0 from the table, we can add that as well as an already unsupported release superseded by 1.2.1.

**Comment by adoroszlai** [comment]
> I think we also miss 1.2.0 from the table, we can add that as well as an already unsupported release superseded by 1.2.1.

@fapifta 1.2.0 was removed due to [my comment](https://github.com/apache/ozone/pull/6100#issuecomment-1911506609).  That release was pulled from the website completely, even the 1.2.0 release page redirects to 1.2.1.  So not only is it unsupported, it is unavailable.  Should we still list it here?

**Comment by fapifta** [comment]
Hmm... sorry for my ignorance, I did not know that 1.2.0 was that seriously purged from everywhere... in this case we should not put back it in here either, you are right.

File: SECURITY.md [hunk 1]
```diff
@@ -5,13 +5,16 @@
...
-| ------------- | ------------------ |
+|---------------| ------------------ |
 | 0.3.0 (alpha) | :x:                |
 | 0.4.0 (alpha) | :x:                |
 | 0.4.1 (alpha) | :x:                |
 | 0.5.0 (beta)  | :x:                |
-| 1.0           | :white_check_mark: |
-| 1.1           | :white_check_mark: |
+| 1.0.0         | :white_check_mark: |
+| 1.1.0         | :white_check_mark: |
```
Comment (adoroszlai) [inline SECURITY.md:14]:
```suggestion
| 1.0.0         | :x: |
| 1.1.0         | :x: |
```

File: SECURITY.md [hunk 2]
```diff
@@ -5,13 +5,16 @@
...
+|---------------| ------------------ |
 | 0.3.0 (alpha) | :x:                |
 | 0.4.0 (alpha) | :x:                |
 | 0.4.1 (alpha) | :x:                |
 | 0.5.0 (beta)  | :x:                |
-| 1.0           | :white_check_mark: |
-| 1.1           | :white_check_mark: |
+| 1.0.0         | :x: |
+| 1.1.0         | :x: |
+| 1.2.1         | :white_check_mark: |
```
Comment (adoroszlai) [inline SECURITY.md:15]:
```suggestion
| 1.0.0         | :x:                |
| 1.1.0         | :x:                |
| 1.2.1         | :white_check_mark: |
```

**Review by fapifta — COMMENTED** [review]
Hi @adoroszlai, @ivandika3,

considering the recent CVE, that affects 1.2.1 and 1.3.0 should we consider removing those also from supported versions, and leave 1.4.0 as supported?

Maybe an other option is to keep 1.3.0 supported, I am not sure if that would create an obligation to release 1.3.1 with the fix for the CVE, or we can leave it as it is?

File: SECURITY.md [hunk 3]
```diff
@@ -5,13 +5,16 @@
...
 | 0.3.0 (alpha) | :x:                |
 | 0.4.0 (alpha) | :x:                |
 | 0.4.1 (alpha) | :x:                |
 | 0.5.0 (beta)  | :x:                |
-| 1.0           | :white_check_mark: |
-| 1.1           | :white_check_mark: |
+| 1.0.0         | :x:                |
+| 1.1.0         | :x:                |
+| 1.2.1         | :white_check_mark: |
+| 1.3.0         | :white_check_mark: |
```
Comment (adoroszlai) [inline SECURITY.md:16]:
```suggestion
| 1.2.1         | :x:                |
| 1.3.0         | :x:                |
```

**Comment by adoroszlai** [comment]
Thanks @ivandika3 for the patch, @fapifta for the review.

> considering the recent CVE, that affects 1.2.1 and 1.3.0 should we consider removing those also from supported versions, and leave 1.4.0 as supported?

I agree, patch is updated accordingly.

I think anyone proposing support for 1.3.x should start by backporting the security fix and volunteering to be release manager.

**Comment by ivandika3** [comment]
Thank you @adoroszlai @fapifta for the in-depth discussions here and in the dev mailing list. I learnt quite a bit about Ozone security policy.

**Pull request merged**